### PR TITLE
Feature/asv 1197 new appc campaigns api

### PR DESCRIPTION
**Pull request by paquiquinho**
**What does this PR do?**

This PR aims to change the retrieval of appc campaign apps from a v7 service to a v8 service; 

**Database changed?**

No

**Where should the reviewer start?**

- [x] CampaignsService.java

**How should this be manually tested?**

Check that the "Get rewarded with appc" bundle still appears in home as well as the more fragment related to that bundle. These apps aren't clickable yet as that's in the scope of another ticket;

**What are the relevant tickets?**

  Tickets related to this pull-request: [ASV-1197](<https://aptoide.atlassian.net/browse/ASV-1197>)

**Code Review Checklist**

- [x] Architecture
- [x] Documentation on public interfaces
- [x] Database changed?
- [x] If yes - Database migration?
- [x] Remove comments & unused code & forgotten testing Logs
- [x] Codestyle
- [x] Partners build
- [x] Unit tests pass
- [x] Functional QA tests pass

### CODE REVIEW

File: app/src/main/java/cm/aptoide/pt/ads/CampaignsService.java [hunk 1]
```diff
@@ -0,0 +1,253 @@
...
+    List<CampaignsServiceResponse.Campaign> campaigns = new ArrayList<>();
+    boolean hasError;
+    String next;
+    if (!response.isSuccessful() || response.body() == null) {
+      hasError = true;
+      next = "";
+    } else {
+      hasError = false;
+      next = response.body()
+          .getNext();
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/ads/CampaignsService.java]: fixed

File: app/src/main/java/cm/aptoide/pt/ads/CampaignsService.java [hunk 2]
```diff
@@ -0,0 +1,253 @@
...
+  private CampaignsServiceV8 service;
+
+  public CampaignsService(CampaignsServiceV8 service) {
+    this.service = service;
+  }
+
+  public Observable<CampaignsServiceResponse> getCampaigns() {
+    return service.getCampaigns()
+        .map(this::mapToResponse)
+        .observeOn(Schedulers.io());
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/ads/CampaignsService.java]: fixed

File: app/src/main/java/cm/aptoide/pt/ads/CampaignsService.java [hunk 3]
```diff
@@ -0,0 +1,253 @@
...
+      }
+    }
+    return new CampaignsServiceResponse(next, campaigns, hasError);
+  }
+
+  public interface CampaignsServiceV8 {
+    @GET("advertising/8.20181126/apps") Observable<Response<ResponseBody>> getCampaigns();
+  }
+
+  public static class ResponseBody {
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/ads/CampaignsService.java]: fixed

File: app/src/main/java/cm/aptoide/pt/ads/CampaignsServiceResponse.java [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+  private String next;
+  private List<Campaign> campaigns;
+
+  public CampaignsServiceResponse(String next, List<Campaign> campaigns, boolean hasError) {
+    this.next = next;
+    this.campaigns = campaigns;
+    this.hasError = hasError;
+  }
+
+  public boolean isHasError() {
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/ads/CampaignsServiceResponse.java]: fixed

File: app/src/main/java/cm/aptoide/pt/home/GridAppCoinsRewardAppsDisplayable.java [hunk 5]
```diff
@@ -45,7 +43,7 @@ public String getTag() {
   }
 
   public void openAppView() {
-    appNavigator.navigateWithAppId(getPojo().getAppId(), getPojo().getPackageName(),
-        AppViewFragment.OpenType.OPEN_ONLY, tag);
+    //appNavigator.navigateWithAppId(getPojo().getAppId(), getPojo().getPackageName(),
+    //    AppViewFragment.OpenType.OPEN_ONLY, tag);
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/home/GridAppCoinsRewardAppsDisplayable.java]: We will have a different base branch instead, as discussed

File: app/src/main/java/cm/aptoide/pt/app/view/MoreBundleFragment.java [hunk 6]
```diff
@@ -281,6 +282,10 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
     toolbar.setLogo(R.drawable.logo_toolbar);
   }
 
+  @Override public void showRewardAppSnack() {
+    Snackbar.make(this.getView(), "Reward app clicked!", Snackbar.LENGTH_LONG);
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/app/view/MoreBundleFragment.java]: fixed

File: app/src/main/java/cm/aptoide/pt/app/view/MoreBundlePresenter.java [hunk 7]
```diff
@@ -121,7 +123,14 @@ private void handleError(HomeBundlesModel.Error error) {
...
               Application app = click.getApp();
-              homeNavigator.navigateToAppView(app.getAppId(), app.getPackageName(), app.getTag());
+              if (app instanceof RewardApp) {
+                RewardApp rewardApp = (RewardApp) app;
+                view.showRewardAppSnack();
+              } else {
+                AptoideApp aptoideApp = (AptoideApp) app;
+                homeNavigator.navigateToAppView(aptoideApp.getAppId(), aptoideApp.getPackageName(),
+                    aptoideApp.getTag());
+              }
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/app/view/MoreBundlePresenter.java]: fixed (extracted navigation to different method)

File: app/src/main/java/cm/aptoide/pt/home/HomeFragment.java [hunk 8]
```diff
@@ -388,6 +389,10 @@ public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container,
     promotionsTicker.setVisibility(View.GONE);
   }
 
+  @Override public void showRewardAppSnack() {
+    Snackbar.make(this.getView(), "Reward app clicked!", Snackbar.LENGTH_LONG);
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/home/HomeFragment.java]: fixed

File: app/src/main/java/cm/aptoide/pt/home/HomePresenter.java [hunk 9]
```diff
@@ -233,12 +234,16 @@ private void handleError(HomeBundlesModel.Error error) {
...
+                homeNavigator.navigateWithEditorsPosition(aptoideApp.getAppId(),
+                    aptoideApp.getPackageName(), "", "", aptoideApp.getTag(),
+                    String.valueOf(click.getAppPosition()));
+              } else if (app instanceof RewardApp) {
+                view.showRewardAppSnack();
               } else {
-                homeNavigator.navigateToAppView(app.getAppId(), app.getPackageName(), app.getTag());
+                AptoideApp aptoideApp = (AptoideApp) app;
+                homeNavigator.navigateToAppView(aptoideApp.getAppId(), aptoideApp.getPackageName(),
+                    aptoideApp.getTag());
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/home/HomePresenter.java]: fixed (also extracted navigation to private method)

File: app/src/main/java/cm/aptoide/pt/home/HomeView.java [hunk 10]
```diff
@@ -48,4 +48,6 @@
   void setEllipsizedPromotionsTicker();
 
   void hidePromotionsIcon();
+
+  void showRewardAppSnack();
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/home/HomeView.java]: fixed

File: app/src/main/java/cm/aptoide/pt/repository/request/RewardAppCoinsAppsRepository.java [hunk 11]
```diff
@@ -22,43 +15,28 @@
 public class RewardAppCoinsAppsRepository {
 
   private static final int APPCOINS_REWARD_LIMIT = 30;
-  private OkHttpClient httpClient;
-  private Converter.Factory converterFactory;
-  private BodyInterceptor<BaseBody> bodyInterceptor;
-  private TokenInvalidator tokenInvalidator;
-  private SharedPreferences sharedPreferences;
   private InstallManager installManager;
+  private CampaignsService campaignsService;
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/repository/request/RewardAppCoinsAppsRepository.java]: fixed

File: app/src/main/java/cm/aptoide/pt/repository/request/RewardAppCoinsAppsRepository.java [hunk 12]
```diff
@@ -22,43 +15,28 @@
...
     this.installManager = installManager;
+    this.campaignsService = campaignsService;
   }
 
-  public Observable<List<Application>> getAppCoinsRewardAppsFromHomeMore(boolean refresh) {
-    return new GetAppCoinsCampaignsRequest(
-        new GetAppCoinsCampaignsRequest.Body(0, APPCOINS_REWARD_LIMIT), httpClient,
-        converterFactory, bodyInterceptor, tokenInvalidator, sharedPreferences).observe(refresh)
-        .flatMap(response -> map(response.getList()));
+  public Observable<List<RewardApp>> getAppCoinsRewardAppsFromHomeMore(boolean refresh) {
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/repository/request/RewardAppCoinsAppsRepository.java]: fixed

File: app/src/main/java/cm/aptoide/pt/ads/CampaignsService.java [hunk 13]
```diff
@@ -0,0 +1,253 @@
...
+                app.getDownloads(), app.getPackageInfo()
+                .getName(), app.getRating()
+                .getAverage(), app.getRating()
+                .getTotal(), app.getCampaign()
+                .getAppc(), app.getCampaign()
+                .getUrls()
+                .getClick(), app.getCampaign()
+                .getUrls()
+                .getDownload(), app.getPackageInfo()
+                .getName()));
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/ads/CampaignsService.java]: fixed

File: app/src/main/java/cm/aptoide/pt/ads/CampaignsServiceResponse.java [hunk 14]
```diff
@@ -0,0 +1,101 @@
...
+    public String getPackageName() {
+      return packageName;
+    }
+
+    public float getAverageRating() {
+      return averageRating;
+    }
+
+    public long getTotalRatings() {
+      return totalRatings;
```
[the remark this answers is not in the document]
Comment (paquiquinho) [reply app/src/main/java/cm/aptoide/pt/ads/CampaignsServiceResponse.java]: fixed

File: app/src/main/java/cm/aptoide/pt/repository/request/RewardAppCoinsAppsRepository.java [hunk 15]
```diff
@@ -22,43 +15,28 @@
...
-            app.getAppcoins() != null && app.getAppcoins()
-                .hasBilling(), app.getAppcoins() != null && app.getAppcoins()
-            .hasAdvertising()));
+  private Observable<List<RewardApp>> map(List<CampaignsServiceResponse.Campaign> list) {
+    List<RewardApp> rewardAppsList = new ArrayList<>();
+    for (CampaignsServiceResponse.Campaign campaign : list) {
+      if (!installManager.wasAppEverInstalled(campaign.getPackageName())) {
+        rewardAppsList.add(
+            new RewardApp(campaign.getLabel(), campaign.getIcon(), campaign.getAverageRating(),
+                campaign.getDownloads(), campaign.getPackageName(), campaign.getUid(), "", true,
```
[the remark this answers is not in the document]
Comment (phillipsgoncalves) [reply app/src/main/java/cm/aptoide/pt/repository/request/RewardAppCoinsAppsRepository.java]: I don't remember any reason on why this should be empty. For this reason, after discussing it, the team decided to create a ticket to review this issue.
https://aptoide.atlassian.net/browse/ASV-1364

**Pull request merged**